firebase-installations/src/main/java/com/google/firebase/installations/remote/FirebaseInstallationServiceClient.java

```diff
@@ -134,43 +134,53 @@ public InstallationResponse createFirebaseInstallation(
                 apiKey));
     while (retryCount <= MAX_RETRIES) {
       HttpsURLConnection httpsURLConnection = openHttpsURLConnection(url);
-      httpsURLConnection.setRequestMethod("POST");
-      httpsURLConnection.setDoOutput(true);
 
-      // Note: Set the iid token header for authenticating the Instance-ID migrating to FIS.
-      if (iidToken != null) {
-        httpsURLConnection.addRequestProperty(X_ANDROID_IID_MIGRATION_KEY, iidToken);
-      }
-
-      GZIPOutputStream gzipOutputStream =
-          new GZIPOutputStream(httpsURLConnection.getOutputStream());
       try {
-        gzipOutputStream.write(
-            buildCreateFirebaseInstallationRequestBody(fid, appId).toString().getBytes("UTF-8"));
-      } catch (JSONException e) {
-        throw new IllegalStateException(e);
-      } finally {
-        gzipOutputStream.close();
-      }
+        httpsURLConnection.setRequestMethod("POST");
+        httpsURLConnection.setDoOutput(true);
 
-      int httpResponseCode = httpsURLConnection.getResponseCode();
+        // Note: Set the iid token header for authenticating the Instance-ID migrating to FIS.
+        if (iidToken != null) {
+          httpsURLConnection.addRequestProperty(X_ANDROID_IID_MIGRATION_KEY, iidToken);
+        }
 
-      if (httpResponseCode == 200) {
-        return readCreateResponse(httpsURLConnection);
-      }
+        writeToOutputStream(httpsURLConnection, fid, appId);
 
-      if (httpResponseCode == 429 || (httpResponseCode >= 500 && httpResponseCode < 600)) {
-        retryCount++;
-        continue;
-      }
+        int httpResponseCode = httpsURLConnection.getResponseCode();
 
-      // Return empty installation response with BAD_CONFIG response code after max retries
-      return InstallationResponse.builder().setResponseCode(ResponseCode.BAD_CONFIG).build();
+        if (httpResponseCode == 200) {
+          return readCreateResponse(httpsURLConnection);
+        }
+
+        if (httpResponseCode == 429 || (httpResponseCode >= 500 && httpResponseCode < 600)) {
+          retryCount++;
+          continue;
+        }
+
+        // Return empty installation response with BAD_CONFIG response code after max retries
+        return InstallationResponse.builder().setResponseCode(ResponseCode.BAD_CONFIG).build();
+      } finally {
+        httpsURLConnection.disconnect();
+      }
     }
 
     throw new IOException();
   }
 
+  private void writeToOutputStream(
```

Comment:
Please make the name describe what is being written to the output stream.

Also, why did you factor it out?

Reply:
Updated.